pkg/acpi/bios.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package acpi
+
+import "fmt"
+
+const (
+	llDSDTAddr = 140
+	lDSDTAddr  = 40
+)
+
+// BiosTable contains the information needed to create table images for
+// firmware such as coreboot or oreboot. It hence includes the RSDP,
+// [XR]SDT, and the raw table data. The *SDT is always Tables[0]
+// as in the real tables.
+type BiosTable struct {
+	RSDP   *RSDP
+	Tables []Table
+}
+
+// ReadBiosTables reads tables that are not interpreted by the OS,
+// i.e. it goes straight to memory and gets them there. We optimistically
+// hope the bios has not stomped around in low memory messing around.
+func ReadBiosTables() (*BiosTable, error) {
```

Comment:
How is this different from ReadTable("mem")? because it includes the RSDP?

Reply:
it goes into and grabs the [RX]DST, and it further can dig deeper into the FADT and get the DSDT, which we currently don't do in the mem case (see comment below).